GLMakie/src/GLAbstraction/AbstractGPUArray.jl

```diff
@@ -61,12 +61,13 @@ function setindex!(A::GPUArray{T, N}, value::Array{T, N}, ranges::UnitRange...)
     nothing
 end
 
+ShaderAbstractions.switch_context!(A::GPUArray) = switch_context!(A.context)
 function update!(A::GPUArray{T, N}, value::AbstractArray{T2, N}) where {T, N, T2}
     update!(A, convert(Array{T, N}, value))
 end
 function update!(A::GPUArray{T, N}, value::AbstractArray{T, N}) where {T, N}
     @sync begin
```

Comment:
Does this actually do anything?

>help?> @sync
  Wait until all lexically-enclosed uses of @async, @spawn, @spawnat and @distributed are complete.

I think that should only wait for anything, if you have an `@async` in the same scope

Reply:
Yeah, I guess we need to make sure that this doesn't happen..
I'm actually not sure if can assure this besides erroring when we switch the context, and then making sure that all our code is written in a way that it doesn't switch...
Ideally, we'd also use multiple threads and then have one context per thread...But that'd be a more advanced refactor 😅

Reply:
I removed all the `@sync`s and tested with 4 windows (2 figures each displaying fig and ax.scene) and it seemed to work fine...

Reply:
I already tried to make sure there are no yields in the rendering code, since that also would crash without multi windows...so maybe that's enough